core/src/block.rs

```diff
@@ -187,6 +187,15 @@ impl<'x> OpenBlock<'x> {
         Ok(())
     }
 
+    pub fn prepare_block_from_transactions(
+        &mut self,
+        block_executor: &dyn BlockExecutor,
+        transactions: &[&TransactionWithMetadata],
```

Comment:
How about making this function receives the `Iteration` instead of the slice of the reference??

Reply:
You mean `Iterator`, right?
I'll change the function.
I'm not sure if it's good to use the `Iterator` type in `TxFilter` and `BlockExecutor` methods (because the traits are high-level interfaces exposed to Coordinator) so I didn't change them yet.
Please let me know if you think it's better to apply the change.

Reply:
Because the `Iterator` is a more abstracted type than the slice, I couldn't understand your concerns.

Reply:
Did you mean it should be serializable?

Reply:
I was slightly worried because the coordinator is directly tied to FML system, and it seemed that it wouldn't be easy to manipulate `Iterator` types in the FML system.

However, as a second thought, if we assume that the coordinator and the host are built and run together in the same process, it wouldn't be problematic to pass `Iterator` type from the host to the coordinator, and it will just be the coordinator's task to unpack the iterator.

In summary, never mind, I'll use iterator type (also in `filter_transactions` method)